### DISCUSSION TITLE
Invert concept of nesting divs

### DISCUSSION THREAD

**Discussion by Brixy** [jgm/djot#299] [Ideas]
Hi all,

This is not a proposal, but rather a suggestion or question.

Nesting divs (and later maybe arbitrary html tags) using different numbers of colons is already possible.

I have the gut feeling that inverting this makes more sense, e.g.

```djot
:::
content

::::
content
::::

:::::
content
:::::
::::
:::
```

Some reasons come to my mind:

**1\.** It has the same feeling as indenting other elements like

```djot
* Item

	* Nested item
```

or asciidoctor.

```asciidoctor
* Item
** Nested Item
```

**2\.** It relates to headings with more `#`s having a lower priority.

```djot
# Heading level 1

## Heading level 2

### Heading level 3
```

**3\.** It’s easier to write when you don’t know man many nesting levels will be required. You just always start with `:::`.

**4\.** Using sidemarks would have the same logic as lists.

```djot
: Item

:: Nested item

::: Nested at a third level
```

Hope this makes sense.

---

Thanks again for djot. I have written an SSG (deno, vento, djot) which is used professionally although not yet being public. We could drop

```
markdown-it
markdown-it-anchor
markdown-it-attrs
markdown-it-deflist
markdown-it-footnote
markdown-it-…
```

Only problem is that definition lists don’t allow multiple descriptions for a single term (and vice versa). But this has been discussed intensively here.

**Comment by jgm** [comment]
Those are all great points.  I think the main arguments for the current system (longer fences on outer divs, shorter on inner) is consistency with the rules for fenced code (which must end with a fence as long as the starting fence; hence, you need bigger fences to include code with a string of backticks on a line by itself).  But that argument might be outweighed by the considerations above.

**Comment by dereuromark** [comment]
This came up in [carve](https://github.com/markup-carve/carve/issues/439) as issue as well.
There was indeed a strong argument to the inversion:

- Consistent with headers
- Allows easier adding of nested ones (instead of re-touching all outer ones)
- Allows admonitions easier

With this proposal an admonition always starts like this (3 colons):
```
::: note
Heads up — this is important.

:::: sub
Sub div here
:::: 

:::
```

How we solved it: split the two axes.

- Code fence keeps `>=`. jgm's argument is correct *there* - code is opaque, never nests, and needs to quote arbitrary fence-shaped content.
- Colon fence switched to exact match. Length is a depth tag, not strength. Canonical direction is now inward-widening (`:::` outermost, +1 per level), which is the direction you can write top-down.

Yes, the `:::: note` + bare `:::` trick is **gone**. Under exact matching that inner `:::` matches nothing, so it *opens a child div*, and would require escaping (but thats expected anyway?).

**Comment by jgm** [comment]
> - Allows easier adding of nested ones (instead of re-touching all outer ones)

I think this is neither here nor there. The inverted approach makes it easier to add new divs inside an existing one. The current approach makes it easier to add new divs that include an existing one.

**Reply by dereuromark** [reply]
Ok fair.
Well the approach outlined here allows for both directions right now with one as default/canonical. 
So could also be then a users choice maybe.

**Comment by vassudanagunta** [comment]
If readability > writability, then


```markdown
::::::::::::::
content

::::
content
::::

:::::
content
:::::
::::
::::::::::::::

```

is far easier to read. Also, the stray div marker on the second-to-last line is easier spot.

**Reply by dereuromark** [reply]
Sure, but that's just what someone is used to. It would be as readable the other way around.
Plus, people would see that as more consistent with also the H nesting depth:

```
# header

::: div

:::::: subdiv
text
::::::

:::

## Subheader

### subsub
```

You are free to double the chars on every sub-div already afaik.
